scripts/monitoring/notification_state.py

```diff
@@ -0,0 +1,27 @@
+import json
+import os
+
+
+class NotificationState:
+    def __init__(self) -> None:
+        with open(os.path.join(os.path.dirname(__file__), "breadcrumbs.json"), "r") as f:
+            self.notified_state = json.load(f)
+
+    def is_set(self, request_id: str, notified_on: str) -> bool:
+        breadcrumb = self.notified_state.get(str(request_id))
+        if breadcrumb is None:
+            return False
+        else:
+            return breadcrumb.get(notified_on, False)
+
+    def update(self, request_id: str, notified_on: str) -> None:
+        breadcrumb = self.notified_state.get(str(request_id))
+        if breadcrumb is None:
+            self.notified_state[str(request_id)] = dict()
+            self.notified_state[str(request_id)][notified_on] = True
```

Comment:
Why are we calling `str(request_id)` if `request_id` is already of type `str`, per function signature?

Also, we should use dict literal `{}` instead of `dict()` when creating empty dictionaries.